src/dotnet-interactive-vscode/src/tests/unit/acquisition.test.ts

```diff
@@ -0,0 +1,333 @@
+        return function(dotnetPath: string, globalStoragePath: string): Promise<string | undefined> {
+            return new Promise<string | undefined>((resolve, reject) => {
+                resolve(version);
+            });
+        };
+    }
+
+    function createEmptyToolManifest(dotnetPath: string, globalStoragePath: string): Promise<void> {
+        return new Promise((resolve, reject) => {
+            const manifestDir = path.join(globalStoragePath, '.config');
+            fs.mkdirSync(manifestDir);
+            const manifestPath = path.join(manifestDir, 'dotnet-tools.json');
+            const manfiestContent = {
+                version: 1,
+                isRoot: true,
+                tools: {}
+            };
+            fs.writeFile(manifestPath, JSON.stringify(manfiestContent), () => resolve());
+        });
+    }
+
+    function createToolManifestThatThrows(dotnetPath: string, globalStoragePath: string): Promise<void> {
+        throw new Error('This function should have never been called.');
+    }
+
+    function installInteractiveTool(args: InstallInteractiveArgs, globalStoragePath: string): Promise<void> {
+        return new Promise((resolve, reject) => {
+            const manifestPath = path.join(globalStoragePath, '.config', 'dotnet-tools.json');
+            fs.readFile(manifestPath, (err, data) => {
+                let manifestContent = JSON.parse(data.toString());
+                manifestContent.tools['microsoft.dotnet-interactive'] = {
+                    version: args!.toolVersion,
+                    commands: [
+                        'dotnet-interactive'
+                    ]
+                };
+                fs.writeFile(manifestPath, JSON.stringify(manifestContent), () => resolve());
+            });
+        });
+    }
+
+    function installInteractiveToolWithSpecificVersion(version: string): { (args: InstallInteractiveArgs, globalStoragePath: string): Promise<void> } {
+        const overrideArgs = {
+            dotnetPath: 'dotnet',
+            toolVersion: version
+        };
+        return (args: InstallInteractiveArgs, globalStoragePath: string) => {
+            return installInteractiveTool(overrideArgs, globalStoragePath);
+        };
+    }
+
+    function installInteractiveToolThatAlwaysThrows(args: InstallInteractiveArgs, globalStoragePath: string): Promise<void> {
+        throw new Error('This function should have never been called.');
+    }
+
+    function report() {
+        // noop
+    }
+
+    it("global storage doesn't exist", async () => {
+        // this tests the scenario where the extension launches for the first time and nothing exists
+        await withFakeGlobalStorageLocation(false, async globalStoragePath => {
+            expect(globalStoragePath).to.not.be.a.path(); // sanity check that it doesn't exist
```

Comment:
Should the sanity check be a separate test (of the test infra itself)? It would make this check and others like it in other tests unnecessary.